journals/apps/core/models.py

```diff
@@ -31,3 +34,76 @@ def get_full_name(self):
     @python_2_unicode_compatible
     def __str__(self):
         return str(self.username)
+
+
+class SiteConfiguration(models.Model):
+    """
+    Each site/tenant should have an instance of this model. This model is responsible for
+    providing databased-backed configuration specific to each site.
+    """
+
+    site = models.OneToOneField('wagtailcore.Site', null=False, blank=False, on_delete=models.CASCADE)
+
+    lms_url_root = models.URLField(
+        verbose_name=_('LMS base url'),
+        help_text=_("Root URL of this site's LMS (e.g. https://courses.stage.edx.org)"),
+        null=False,
+        blank=False
+    )
+
+    lms_public_url_root_override = models.URLField(
+        verbose_name=_('LMS public base url'),
+        help_text=_(
+            "Root public URL of this site's LMS. Only used for overwriting client-side URLs"
+        ),
+        null=True,
+        blank=True
+    )
+
+    oauth_settings = JSONField(
+        verbose_name=_('OAuth settings'),
+        help_text=_('JSON string containing OAuth backend settings.'),
+        null=False,
+        blank=False,
+        default={}
+    )
+
+    def build_lms_url(self, path=''):
+        """
+        Returns path joined with the appropriate LMS URL root for the current site.
+
+        Returns:
+            str
+        """
+        return urljoin(self.lms_url_root, path)
+
+    @property
+    def oauth2_provider_url(self):
+        """ Returns the URL for the OAuth 2.0 provider. """
+        return self.build_lms_url('/oauth2')
```

Comment:
Would we want to use this same mechanism to build oauth provide url's to the other services (discovery, ecommerce)?
Currently this is lms specific

Reply:
The LMS is our OAuth provider. It provides authentication to all of the services, so we only need to get the token from LMS, then pass it along to our other API calls.

Reply:
ok, didn't realize that was the case